File: model/rgb.py
def _clamp(val):
    return max(0, min(val, 255))


class RGB:
    r: int
    g: int
    b: int

    def __init__(self, r: int, g: int, b: int):
        self.r = _clamp(int(r))
        self.g = _clamp(int(g))
        self.b = _clamp(int(b))

    def __str__(self):
        return f"({self.r}, {self.g}, {self.b})"

File: stickman/cpp/src/rgb.cpp
#include "../include/rgb.h"

#include <algorithm>
#include <cmath>


int clamp(int val) { return std::min(std::max(val, 0), 255); }

RGB::RGB() {
  red = 0;
  green = 0;
  blue = 0;
}

RGB::RGB(int r, int g, int b) {
  red = clamp(r);
  green = clamp(g);
  blue = clamp(b);
}

RGB::RGB(float r, float g, float b) {
  red = clamp(int(std::round(r)));
  green = clamp(int(std::round(g)));
  blue = clamp(int(std::round(b)));
}

int RGB::as_int() const { return (red << 16) | (green << 8) | blue; }